ament_cmake_ikos/cmake/ament_ikos.cmake

```diff
@@ -19,6 +19,7 @@ function(ament_ikos)
   ament_add_test(
     "${ARG_TESTNAME}"
     COMMAND ${cmd}
+    TIMEOUT 1800
```

Comment:
I'd recommend making this come from an argument so that the timeout can be set on a per-package basis. This allows larger packages to have larger timeouts while not forcing smaller packages to wait a long time if a bug causes the test to hang later.

See `ament_cpplint` for an example: 

* Documentation for TIMEOUT parameter https://github.com/ament/ament_lint/blob/4bd8e5bc0ae0ed891400a20b7e2fbd0835ca05b8/ament_cmake_cpplint/cmake/ament_cpplint.cmake#L27-L28
* Parsing as a one-arg parameter to the CMake function https://github.com/ament/ament_lint/blob/4bd8e5bc0ae0ed891400a20b7e2fbd0835ca05b8/ament_cmake_cpplint/cmake/ament_cpplint.cmake#L37
* Default value setting https://github.com/ament/ament_lint/blob/4bd8e5bc0ae0ed891400a20b7e2fbd0835ca05b8/ament_cmake_cpplint/cmake/ament_cpplint.cmake#L63-L65
* passing to ament_add_test https://github.com/ament/ament_lint/blob/4bd8e5bc0ae0ed891400a20b7e2fbd0835ca05b8/ament_cmake_cpplint/cmake/ament_cpplint.cmake#L74

Reply:
Hi, that is good suggestion. I can definitely parametrize that. However, I would keep the default value to 1800. The reason is that ament_ikos is invoked via ament_lint_common and ament_lint_auto in most of the packages (including rclcpp). ament_ikos is invoked in rclcpp only in spaceros distribution, which patches ament_lint_common to include ikos.
Is that ok?

Reply:
There's a pattern for excluding one of the checks from `ament_lint_auto` and then calling it separately to use non-default arguments.

https://github.com/ros2/rclcpp/blob/1981e1f12acfcbdc5d9202cd3e636cb692621971/rclcpp_lifecycle/CMakeLists.txt#L62-L68


Just checking so I don't push to hard down a difficult path - would modifying the timeout on a per-package basis be a significant amount of work? How many packages do you think would need the ikos timeout increased?

Reply:
I would love to do that but that would be hard, let me explain why.
1. We are using ament_ikos in [spaceros](https://github.com/space-ros/space-ros)
2. The way linting in spaceros works is that we are cloning upstream ros2 packages from standard ros2 version (e.g. `rclcpp`) without modifying them, and we use our own version of `ament_lint_common` to include `ament_ikos` (note that it is not included in standard ros2 distro). 
3. I am not sure if updating upstream packages like `rclcpp` is good idea, because `ament_ikos` is not used in regular ros2 distribution. Also, current approach is quite low-touch and allows to avoid divergence between core ros2 packages and spaceros.

I am working on improving our build system which will allow us to override CMakeLists.txt files in upstream packages during spaceros build, however it will take some time, and for now analysis simply does not work because threshold is too low.  Nevertheless it will allow us to lower the default in the future.

Reply:
Ohhh. I see that it makes sense to not modify rclcpp because rclcpp normally wouldn't depend on `ament_ikos`.

Thank you for your patience. In that case I'm on board with updating the timeout here.